server/routes/api/integrations/integrations.ts

```diff
@@ -127,15 +102,27 @@ router.post(
 router.post(
   "integrations.delete",
   auth({ admin: true }),
-  async (ctx: APIContext) => {
-    const { id } = ctx.request.body;
-    assertUuid(id, "id is required");
-
+  transaction(),
+  validate(T.IntegrationsDeleteSchema),
```

Comment:
```suggestion
  validate(T.IntegrationsDeleteSchema),
  transaction(),
```

Lets make sure transaction is the last middleware